### PR TITLE
Add C# code blocks to the Using JSON guide

### PR DESCRIPTION
**Pull request by Srainyyyy**
Add C# code blocks to the three existing JSON guides to provide equivalent examples for C# developers

### CODE REVIEW

File: src/content/docs/guides/json/0-json_intro.mdx [hunk 1]
```diff
@@ -72,8 +72,26 @@ This command creates sub-folders for each type of resource. One of these is name
   ```
 
   </TabItem>
+
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/0-json_intro.mdx:75]: Need to check the functions you are using as they do not exist.

The below compiles and runs when game_data.json exists in the Resource/json folder within your project:

using SplashKitSDK;

class JsonTutorial
{
    static void Main()
    {
        Json gameData;
        gameData = SplashKit.JsonFromFile("game_data.json");
        string gameTitle = SplashKit.JsonReadString(gameData, "gameTitle");
        
        SplashKit.WriteLine("Game Title: " + gameTitle);
        
        SplashKit.FreeJson(gameData);
    }
}

File: src/content/docs/guides/json/0-json_intro.mdx [hunk 2]
```diff
@@ -137,6 +177,29 @@ But what if we didn't have a `gameTitle` key in our JSON? Well, error messages w
...
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/0-json_intro.mdx:180]: As above, this code does not compile correctly. 

See below:

```
using SplashKitSDK;

class JsonTutorial
{
    static void Main()
    {
        Json gameData;
        gameData = SplashKit.JsonFromFile("game_data.json");

        if(SplashKit.JsonHasKey(gameData, "gameTitle"))
        {
            string gameTitle = SplashKit.JsonReadString(gameData, "gameTitle");
            SplashKit.WriteLine("Game Title: " + gameTitle);
        }
        else
        {
            SplashKit.WriteLine("Key \"gameTitle\" not found.");
        }

        SplashKit.FreeJson(gameData);
    }
}
```

File: src/content/docs/guides/json/1-json_reading.mdx [hunk 3]
```diff
@@ -73,6 +83,28 @@ Below is an example of this:
...
 
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+
+  var gameData = SplashKit.LoadJson("game_data.json");
+
+  var levels = new List<string>();
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/1-json_reading.mdx:93]: Recommend:

`List<string> levels = new List<string>();`

File: src/content/docs/guides/json/1-json_reading.mdx [hunk 4]
```diff
@@ -73,6 +83,28 @@ Below is an example of this:
...
+  SplashKit.JsonReadArray(gameData, "levels", levels);
+
+  int numLevels = levels.Count;
+
+  for (int i = 0; i < numLevels; i++)
+  {
+     SplashKit.WriteLine("Got level: " + levels[i]);
+  }
+
+  SplashKit.FreeJson(gameData);
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/1-json_reading.mdx:104]: Recommend removing this to be in line with the C++ code.

File: src/content/docs/guides/json/1-json_reading.mdx [hunk 5]
```diff
@@ -97,6 +129,16 @@ SplashKit's JSON functionality allows you to extract various types of data, incl
   ```
 
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+  Json gameScreenSize = SplashKit.JsonReadObject(game_data, "screenSize");
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/1-json_reading.mdx:136]: Change the Json variable from game_data to gameData.

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 6]
```diff
@@ -31,6 +31,17 @@ In SplashKit, you can programmatically create JSON objects and arrays, which the
   ```
 
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+  var newGameData = Json.Create();
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:38]: These functions do not exist. 

See below:

Json newGameData = SplashKit.CreateJson();
SplashKit.JsonSetString(newGameData, "gameTitle", "My New Game");
SplashKit.JsonSetBool(newGameData, "fullScreenMode", false);
SplashKit.JsonSetNumber(newGameData, "numPlayers", 1);

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 7]
```diff
@@ -49,6 +60,21 @@ First we create the new JSON object using [Create Json](/api/json/#create-json),
   ```
 
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+  var levelsArray = new List<string>
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:67]: Function does not exist. See below:

```
List<string> levelsArray = new List<string>
    {
        "level1",
        "level2",
        "level3"
	};

SplashKit.JsonSetArray(newGameData, "levels", levelsArray);
```

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 8]
```diff
@@ -66,6 +92,19 @@ Next we add the levels array to the JSON object. We create a vector to store the
   ```
 
   </TabItem>
+
+  <TabItem label="C#">
+  
+  ```csharp
+  var screenSizeData = Json.Create();
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:99]: Functions do not exist. See below:

```
Json screenSizeData = SplashKit.CreateJson();

SplashKit.JsonSetNumber(screenSizeData, "width", 800);
SplashKit.JsonSetNumber(screenSizeData, "height", 600);

SplashKit.JsonSetObject(newGameData, "screenSize", screenSizeData);
```

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 9]
```diff
@@ -82,6 +121,15 @@ Now that we have the `new_game_data` object that stores the same values as the J
   ```
   
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+  newGameData.ToFile("new_game_data.json");
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:128]: Function does not exist. See below:

`SplashKit.JsonToFile(newGameData, "new_game_data.json");`

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 10]
```diff
@@ -122,6 +170,32 @@ By combining all these examples we can create the full program shown below.
   ```
 
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:176]: See above comments and adjust code to ensure these are using the correct C# functions.

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 11]
```diff
@@ -165,6 +239,22 @@ You can also load an existing JSON file, modify its contents, and save the chang
   ```
   
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+  var playerData = Json.Create();
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:246]: As above - check that these are the correct C# functions.

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 12]
```diff
@@ -180,6 +270,17 @@ First we create the player JSON object to store the data for an entire character
   ```
 
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+  var existingData = Json.FromFile("new_game_data.json");
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:277]: As above - check that these are the correct C# functions.

File: src/content/docs/guides/json/2-json_writing.mdx [hunk 13]
```diff
@@ -226,6 +327,21 @@ Now we have a character object stored with this JSON file. We also now have mult
   ```
   
   </TabItem>
+
+  <TabItem label="C#">
+
+  ```csharp
+  // Load our JSON
+  var modifiedGameData = Json.FromFile("modified_game_data.json");
```
Comment (ShaunR1991) [inline src/content/docs/guides/json/2-json_writing.mdx:335]: As above - check that these are the correct C# functions.

**Review by ShaunR1991 — COMMENTED** [review]
This code needs to be re-done as very many of these functions are not the correct C# functions and thus the code will not compile. Ensure code is tested before reuploading.

**Review by ShaunR1991 — APPROVED** [review]
All code compiles and is the C# facsimile of the C++ code. Looks good!

**Review by s223126445 — APPROVED** [review]
All Codes were running. Used the top-level statements in C# code and maintained the structure as C++.
yang1
yang2
yang3
yang4

**Review by omckeon — CHANGES_REQUESTED** [review]
For all code blocks:

Please add `using static SplashKitSDK.SplashKit;` below line 1.

Then remove all the `SplashKit.` parts of the code.

**Review by omckeon — APPROVED** [review]
Thank you for these updates. Code blocks look great now!

**Comment by omckeon** [comment]
Have merged this now, and will merge it upstream as well :)

Just wanted to note that for future pull requests, make sure you have created a new branch to add the changes into, rather than to main. And make sure to pull the changes from upstream into your fork as well :)

**Pull request merged**